mobile/src/main/java/ch/epfl/sdp/mobile/ui/puzzles/PuzzleGameScreenState.kt

```diff
@@ -0,0 +1,15 @@
+package ch.epfl.sdp.mobile.ui.puzzles
+
+import androidx.compose.runtime.Stable
+import ch.epfl.sdp.mobile.ui.game.ChessBoardState
+
+/**
+ * A state which indicates the content of a [PuzzleGameScreen] composable. It will keep track of the
+ * values of moves history.
+ *
+ * @param Piece the type of the pieces of the underlying [ChessBoardState].
+ */
+@Stable
+interface PuzzleGameScreenState<Piece : ChessBoardState.Piece> {
```

Comment:
Question : why is the generic `Piece` type required ?

Reply:
It is not needed right now, but will be later (another PR). 
For now, it is simplified in [1b071fe](https://github.com/epfl-SDP/android/pull/300/commits/1b071fe17a24974a75f8962233fa858ce8d880f6)

Reply:
Indeed, I misread the interface name. Sorry !